e2e/e2e_test.go

```diff
@@ -551,3 +552,100 @@ func TestNamespaceIssuers(t *testing.T) {
 		}
 	}
 }
+
+func TestCertificateRecovery(t *testing.T) {
```

Comment:
Can we make this test name long, descriptive, and possibly ugly? I want it to be ultra-clear what this test is validating.